api/app/Http/Controllers/Api/PasswordResetController.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use Mail;
+use App\User;
+use Carbon\Carbon;
+use App\PasswordReset;
+use Webpatser\Uuid\Uuid;
+use Illuminate\Http\Request;
+use App\Mail\PasswordResetMail;
+use App\Http\Controllers\Controller;
+use App\Http\Requests\Api\NewPasswordModel;
+use App\Http\Requests\Api\PasswordResetModel;
+
+class PasswordResetController extends Controller
+{
+    /**
+     * Send a mail for resetting the password.
+     *
+     * @param \App\Http\Requests\Api\PasswordResetModel $request
+     *
+     * @return mixed
+     */
+    public function sendResetMail(PasswordResetModel $request)
+    {
+        $userEmail = $request->email;
+        $user = User::where('email', $userEmail)->first();
+        if ($user && ! $this->isSpamming($user, config('app.password_reset_minutes'))) {
+            // User exists and had no request < app.password_reset_minutes
+            $token = Uuid::generate(4).'-'.str_random(40);
+            $this->sendPasswordResetMail([
+                'email' => $userEmail,
+                'url' => url('/reset-password/'.$token),
+                'name' => $user->name,
+            ]);
+            // Store in database
+            $data = [
+                'user_id' => $user->id,
+                'token' => $token,
+                'created_at' => Carbon::now(),
+            ];
+            PasswordReset::create($data);
+        }
+    }
+
+    private function isSpamming(User $user, $minutes)
+    {
+        if ($user) {
+            $last_password_request = PasswordReset::where('user_id', $user->id)->orderBy('created_at', 'desc')->first();
```

Comment:
Use Camel Case

Use Eloquent relationships: User::with('passwordresets'), so we can do it all in one query.